glnx-fdio.c

```diff
@@ -741,6 +724,37 @@ glnx_regfile_copy_bytes (int fdf, int fdt, off_t max_bytes, gboolean try_reflink
             m = (size_t) max_bytes;
         }
 
+      /* First try copy_file_range(), unless we already tried.  Note this is an
+       * inlined version of try_copy_file_range() from systemd upstream, which
+       * works better since we use POSIX errno style.
+       */
+      if (try_cfr)
+        {
+          n = copy_file_range (fdf, NULL, fdt, NULL, m, 0u);
+          if (n < 0)
+            {
+              if (errno == ENOSYS)
+                {
+                  have_cfr = 0;
+                  try_cfr = false;
```

Comment:
And `can_copy_file_range()` would do a dummy syscall?  Feels uglier to me offhand.